charts/k8ssandra/templates/grafana/configmap.yaml

```diff
@@ -0,0 +1,16 @@
+{{- if .Values.monitoring.grafana.provision_dashboards }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-k8ssandra-grafana-dashboards
+  labels:
+    grafana_dashboard: "true"
```

Comment:
@bradfordcp you mentioned that Grafana will look for the `grafana_dashboard: true` label. Do you know if we can configure Grafana to filter on additional labels as well. The answer may have a significant impact on what deployment topologies we can support, e.g., one vs multiple k8ssandra releases per namespace.

Reply:
It looks like we have a few options here. I'm not a fan of any one in particular, but here we go:

1. Sidecar - _currently used_ - We provide a configmap with the dashboards within it. The sidecar looks for a specific label key, but ignores the value. The default is `grafana_dashboard`. See https://github.com/grafana/helm-charts/blob/main/charts/grafana/templates/_pod.tpl#L86 and https://github.com/kiwigrid/k8s-sidecar#configuration-environment-variables for more info. It may be worth our time to put in an issue and PR for the upstream helm chart to allow for setting the value or default to something like matching the Helm release name (which `kube-prometheus-stack` does for service monitors)
2. Embedded - We could include the JSON or a file path a la https://github.com/grafana/helm-charts/blob/main/charts/grafana/templates/dashboards-json-configmap.yaml. Embedding JSON is pretty ugly as we're talking about thousands of lines of dashboard content in values.yaml. The other option, specifying a filename to be included doesn't work with subcharts (see https://github.com/helm/helm/issues/1892). Which is unlikely to be resolved anytime soon.
3. Download - There is support for specifying a URL that is downloaded when Grafana starts up. We _could_ point this at the MCAC repo, but that assumes the cluster has internet access and seems like a less than stellar approach.

Reply:
@jsanda @burmanm I'd love to hear your thoughts on these solutions.

Reply:
I am firmly opposed to 3. Option 2 isn't great, but doable. If I understand correctly, the JSON would be added when the template engine runs which seems ok. Option 1 sounds like the best long term, but the label situation is improved, maybe option 2 is our best bet?

Reply:
Option 2 is firmly blocked by the Helm issue linked unless we want to dump all of our dashboard JSON into values.yaml (I'm a :-1:  on that approach). It would be static inside of values.yaml.